src/ferui-components/tree-view/tree-node-component.ts

```diff
@@ -0,0 +1,168 @@
+          </ng-template>
+        </span>
+      </div>
+    </div>
+    <div [style.margin-left.px]="calculatePadding() + 20">
+      <clr-icon *ngIf="node.showLoader" class="fui-loader-animation" shape="fui-spinner"></clr-icon>
+      <clr-icon *ngIf="node.loadError" class="fui-error-icon" shape="fui-error" aria-hidden="true"></clr-icon>
+      <span *ngIf="node.loadError" class="error-msg">Couldn't load content</span>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FuiTreeNodeComponent<T> implements OnInit, DoCheck {
+  @Output() onNodeEvent: EventEmitter<TreeNodeEvent<T>> = new EventEmitter<TreeNodeEvent<T>>();
+
+  @Input() node: TreeNode<T>;
+
+  @Input() theme: TreeViewColorTheme;
+
+  @Input() dataRetriever: TreeNodeDataRetriever<T> | PagedTreeNodeDataRetriever<T>;
+
+  @HostBinding('class') themeClass;
+
+  @HostBinding('class.fui-tree-node-component') nodeComponent: boolean = true;
+
+  @HostBinding('class.borders') @Input() borders: boolean = false;
+
+  @ViewChild('nodetree', { read: ElementRef }) nodeTreeElement: ElementRef;
+
+  // Hierarchical level to show parent-child relationship
+  level: number = 0;
+  // Indicates node can be expanded
+  hasChildren: boolean;
+  // left padding dependent of node tree hierarchical level
+  padding: number;
+
+  constructor(private cd: ChangeDetectorRef) {}
+
+  /**
+   * Initiates Tree Node component and awaits to see if node has children while also setting its hierchical level
+   * based on the number of parents it has
+   *
+   * @returns {Promise<void>}
+   */
+  async ngOnInit() {
+    this.themeClass = this.theme.toLowerCase();
+    this.hasChildren = await this.dataRetriever.hasChildNodes(this.node.data);
+    let parent = this.node.parent;
+    while (parent != null) {
+      parent = parent.parent;
+      this.level++;
+    }
+    this.padding = this.calculatePadding();
+    this.cd.markForCheck();
+  }
+
+  /**
+   * Checks to ensure node is updated
+   */
+  ngDoCheck(): void {
+    this.cd.markForCheck();
+  }
+
+  /**
+   * Gets the Node Tree width including padding
+   * @returns {number}
+   */
+  getNodeWidth(): number {
+    return this.nodeTreeElement.nativeElement.offsetWidth;
+  }
+
+  /**
+   * Invokes the node event based on the host Tree Node and its expanded or collapsed state
+   */
+  onExpand(): void {
+    this.onNodeEvent.emit({
+      getNode: () => {
+        return this.node;
+      },
+      getType: () => {
+        return !this.node.expanded ? TreeViewEventType.NODE_EXPANDED : TreeViewEventType.NODE_COLLAPSED;
+      },
+    });
+    this.cd.markForCheck();
+  }
+
+  /**
+   * Invokes the node event based on the host Tree Node click event
+   */
+  onSelected(): void {
+    this.onNodeEvent.emit({
+      getNode: () => {
+        return this.node;
+      },
+      getType: () => {
+        return TreeViewEventType.NODE_CLICKED;
+      },
+    });
+    this.cd.markForCheck();
```

Comment:
You're not updating any variables, so no need to call `this.cd.markForCheck();` there.